### PR TITLE
Issue #19: add permission check to the settings page

### PR DESCRIPTION
**Pull request by dmitriim**

### CODE REVIEW

**Comment by dmitriim** [comment]
This is pretty much a copy from object fs code.

File: lang/en/tool_s3logs.php [hunk 1]
```diff
@@ -52,11 +52,14 @@
...
+$string['permissioncheckpassed'] = 'Permissions check passed.';
 $string['privacy:metadata'] = 's3logs tool export Moodle standard log for archiving purposes';
 $string['privacy:metadata:tool_s3logs:realuserid'] = 'The ID of the real user behind the event, when masquerading a user.';
 $string['privacy:metadata:tool_s3logs:relateduserid'] = 'The ID of a user related to an event';
 $string['privacy:metadata:tool_s3logs:userid'] = 'The ID of the user who triggered an event';
 $string['privacy:metadata:tool_s3logs:externalpurpose'] = 's3logs tool exports Moodle standard log records for archiving purposes';
 $string['usesdkcreds'] = 'Use the default credential provider chain to find AWS credentials';
 $string['usesdkcreds_desc'] = 'If Moodle is hosted inside AWS, the default credential chain can be used for access to s3 logs.';
+$string['writefailure'] = 'Could not write object to the S3 storage. {$a}';
+
```
Comment (golenkovm) [inline lang/en/tool_s3logs.php]: Accidentally added an empty string?

File: classes/local/client/s3_client.php [hunk 2]
```diff
@@ -133,6 +133,35 @@ public function upload_file(string $filepath, string $keyname): ?string {
...
+     *
+     * @return string
+     */
+    public function get_client_status_message(): string {
+        global $OUTPUT;
+
+        $output = '';
+        $connection = $this->test_connection();
+        if ($connection->success) {
+            $output .= $OUTPUT->notification(get_string('connectionsuccess', 'tool_s3logs'), 'notifysuccess');
```
Comment (golenkovm) [inline classes/local/client/s3_client.php:147]: As I know, `notifysuccess` and `notifyproblem` are considered as legacy type of notification, see https://github.com/moodle/moodle/blob/master/lib/outputrenderers.php#L3016-L3029
I would swap to `success` and `error` respectively

File: classes/local/client/s3_client.php [hunk 3]
```diff
@@ -164,6 +193,41 @@ public function test_connection() {
...
+     */
+    public function test_permissions() {
+        $permissions = new \stdClass();
+
+        $permissions->success = true;
+        $permissions->messages = [];
+
+        if ($this->is_functional()) {
+            $permissions->success = false;
+            $permissions->messages = [];
```
Comment (golenkovm) [inline classes/local/client/s3_client.php:209]: Do we need to add a message here that it's not functional?

File: classes/local/client/s3_client.php (hunk 2 above)
Comment (dmitriim) [reply classes/local/client/s3_client.php:147]: Good catch. Let's fix  them all in once as a separate PR.

Comment (dmitriim) [reply classes/local/client/s3_client.php:147]: https://github.com/catalyst/moodle-tool_s3logs/issues/23

File: classes/local/client/s3_client.php (hunk 3 above)
Comment (dmitriim) [reply classes/local/client/s3_client.php:209]: This should be covered by test connection. So we don't need messages here.

File: lang/en/tool_s3logs.php (hunk 1 above)
Comment (dmitriim) [reply lang/en/tool_s3logs.php]: fixed

**Pull request merged**